sql/sql_repl.cc

```diff
@@ -1861,7 +1874,7 @@ send_event_to_slave(binlog_send_info *info, Log_event_type event_type,
       if (until_gtid_state)
       {
         gtid= until_gtid_state->find(event_gtid.domain_id);
-        if (gtid == NULL)
+        if (gtid == NULL && !info->is_until_before_gtids)
```

Comment:
Minor style suggestion:
I would add the new condition inside an extra if(!info->is_until_before_gtids){...} ...